src/routes/assayclassifier/__init__.py

```diff
@@ -80,90 +78,94 @@ def calculate_assay_info(metadata: dict,
         return {}
 
 
-def calculate_data_types(entity: dict) -> list[str]:
+def calculate_data_types(entity_json: dict) -> list[str]:
     data_types = [""]
 
     # Historically, we have used the data_types field. So check to make sure that
     # the data_types field is not empty and not a list of empty strings
     # If it has a value it must be an old derived dataset so use that to match the rules
-    if hasattr(entity, "data_types") and entity.data_types \
-            and set(entity.data_types) != {""}:
-        data_types = entity.data_types
+    if ("data_types" in entity_json and entity_json["data_types"]
+            and set(entity_json["data_types"]) != {""}):
+        data_types = entity_json["data_types"]
     # Moving forward (2024) we are no longer using data_types for derived datasets.
     # Rather, we are going to use the dataset_info attribute which stores similar information
     # to match the rules. dataset_info is delimited by "__", so we can grab the first
     # item when splitting by that delimiter and pass that through to the rules.
-    elif hasattr(entity, "dataset_info") and entity.dataset_info:
-        data_types = [entity.dataset_info.split("__")[0]]
+    elif "dataset_info" in entity_json and entity_json["dataset_info"]:
```

Comment:
could this (and other similar ifs) be simplified to 
`if entity_json.get("dataset_info"):`

Reply:
I actually did it that way to exactly reproduce the earlier logic. Yes, I think it could be simplified as you describe, but I wasn't sure what case the old code was intended to prevent so I left it. I could change it easily enough- @gesinaphillips do you have an opinion?